### ISSUE TITLE
Search: too many results, narrow by keyword

### ISSUE THREAD

**Issue by josh-chamberlain** [Police-Data-Accessibility-Project/data-sources-app#289]
### Context

Sometimes users will make a search with 100+ results, which is often too many no matter how you group them.

This can serve as a precursor to advanced search; if we capture what people use for keywords...could be useful.

### Requirements

- If there are 50 or more results **and** the user already selected one of the record type categories besides "all", we should show: Want to narrow down the results? `Text box "Enter a keyword"` `Submit`

#### API

- We can use similar logic to the quick search endpoint, the "what kind of data" all-purpose field.

**Comment by maxachis** [comment]
@josh-chamberlain Considering our search endpoint now relies on locations, rather than keywords, does this logic still apply?

**Comment by josh-chamberlain** [comment]
@maxachis this idea doesn't conflict with that—we could still use keywords to narrow down a search. You could still create a search for a state x record category that gives 100 results; what if you just want to narrow down by "force" or "muckrock" or "2023" and search every column in the table...but _only_ for the results on page...to narrow it down?

Might not be worth it, but not dead yet!

**Comment by maxachis** [comment]
@josh-chamberlain Got it! I will label this as post-v2, then!

**Comment by maxachis** [comment]
@josh-chamberlain Just to make sure I've got this understood, would this require changes to the frontend, backend, or both? I could see it possibly being done only on the frontend, if we're comfortable with more data than the user will use being provided from the endpoint.

**Comment by josh-chamberlain** [comment]
@maxachis yes, I think front end only will probably be OK—we're filtering what's shown based exclusively on properties which have already been returned from the API.
